util.c

```diff
@@ -1932,6 +1932,55 @@ int mdmon_running(const char *devnm)
 	return 0;
 }
 
+/* wait_for_mdmon_control_socket() - Waits for mdmon control socket
+ * to be created within specified time.
+ * @devnm: Device for which mdmon control socket should start.
+ *
+ * In foreground mode, when mdadm is trying to connect to control
+ * socket it is possible that the mdmon has not created it yet.
+ * Give some time to mdmon to create socket. Timeout set to 10 sec.
+ */
+int wait_for_mdmon_control_socket(char *devname)
+{
+	char path[100];
+	int sfd;
+	struct sockaddr_un addr;
+	int pos;
+	char *c;
+	int rv, retry_count = 0;
+	pos = sprintf(path, "%s/", MDMON_DIR);
+	if (is_subarray(devname)) {
+		devname++;
+		c = strchr(devname, '/');
+		if (!c)
+			return -1;
+		snprintf(&path[pos], c - devname + 1, "%s", devname);
+		pos += c - devname;
+	} else
+		pos += sprintf(&path[pos], "%s", devname);
+	sprintf(&path[pos], ".sock");
+
+	sfd = socket(PF_LOCAL, SOCK_STREAM, 0);
+	if (sfd < 0)
+		return -1;
+
+	addr.sun_family = PF_LOCAL;
+	strcpy(addr.sun_path, path);
+
+	for (retry_count = 0; retry_count < 20; retry_count++) {
```

Comment:
Do we need 10 seconds? Previously, we determined that 2 seconds are enough.

Reply:
10 seconds will block it from being used in other places in the future, especially from udev context. Can we stay with 2 seconds as we previously determined?